### PR TITLE
Shared `libgcc` can now be enabled for `aarch64-w64-mingw32`

### PR DESCRIPTION
**Pull request by Blackhex**
After Windows-on-ARM-Experiments/gcc-woarm64#21 was merged, shared `libgcc` can be enabled for `aarch64-w64-mingw32`.

### CODE REVIEW

**Comment by Blackhex** [comment]
@eukarpov It works now, the only thing to discuss/resolve is whether to copy the DLLs to both locations needed for OpenSSL as it is now, or copy them to some common location and add them to `PATH`.

File: .github/scripts/config.sh [hunk 1]
```diff
@@ -56,6 +56,8 @@ FFMPEG_TESTS_PATH=${FFMPEG_TESTS_PATH:-~/ffmpeg-tests}
 CCACHE_LIB_DIR=/usr/lib/ccache
 TOOLCHAIN_CCACHE_LIB_DIR=$TOOLCHAIN_PATH/lib/ccache
 
+GCC_VERSION="15.0.0"
```
Comment (eukarpov) [inline .github/scripts/config.sh:59]: What about getting the GCC version from here?
gcc/BASE-VER

File: .github/scripts/libjpeg-turbo/pack-tests.sh [hunk 2]
```diff
@@ -6,8 +6,14 @@ LIBJPEG_TURBO_BUILD_PATH=$BUILD_PATH/libjpeg-turbo
 
 echo "::group::Pack libjpeg-turbo tests"
     mkdir -p $ARTIFACT_PATH
-    cd $LIBJPEG_TURBO_BUILD_PATH
-    zip -r $ARTIFACT_PATH/$TOOLCHAIN_NAME-libjpeg-turbo-tests.zip *
+
+    pushd $LIBJPEG_TURBO_BUILD_PATH
+        cp $TOOLCHAIN_PATH/$TARGET/bin/*.dll .
```
Comment (eukarpov) [inline .github/scripts/libjpeg-turbo/pack-tests.sh:11]: I suggest to create a separate package with shared libraries and use it when it is needed for testing.

**Comment by eukarpov** [comment]
> @eukarpov It works now, the only thing to discuss/resolve is whether to copy the DLLs to both locations needed for OpenSSL as it is now, or copy them to some common location and add them to `PATH`.

I think using a common package and some location will be good idea.

File: .github/scripts/libjpeg-turbo/pack-tests.sh (hunk 2 above)
Comment (Blackhex) [reply .github/scripts/libjpeg-turbo/pack-tests.sh]: DLLs moved to a separate artifact package.

**Review by eukarpov — APPROVED** [review]

**Comment by Blackhex** [comment]
Significantly changed and ready for review.

File: .github/scripts/toolchain/build-mingw-winpthreads.sh [hunk 3]
```diff
@@ -16,19 +16,12 @@ if [[ "$RUN_CONFIG" = 1 ]] || [[ ! -f "$MINGW_BUILD_PATH/Makefile" ]]; then
...
-            ;;
-        esac
-
         $SOURCE_PATH/mingw/mingw-w64-libraries/winpthreads/configure \
             --prefix=$TOOLCHAIN_PATH/$TARGET \
             --build=$BUILD \
             --host=$TARGET \
             --enable-static \
-            --enable-shared \
+            --disable-shared \
```
Comment (vejbomar) [inline .github/scripts/toolchain/build-mingw-winpthreads.sh:24]: Can you elaborate please why you had to disable building shared version of `winpthreads`?

**Review by vejbomar — APPROVED** [review]
The PR looks good to me, only a motivation for disabling shared version of `winpthreads` is unclear to me.

File: .github/scripts/toolchain/build-mingw-winpthreads.sh (hunk 3 above)
Comment (Blackhex) [reply .github/scripts/toolchain/build-mingw-winpthreads.sh:24]: I am glad you asked as this is quite important question: The reason is that when the stage 1 GCC is being built, it needs to be built with static `libgcc` only as stage1 compiler is not able to link binaries. Then the runtime, which `winpthreads` is part of, needs to be built but shared `libgcc` is not available yet so it needs to be built statically as well. Then the stage 2 GCC is built with all those dependencies with both static and shared `libgcc` and then `winpthread` is re-built again both static and share in `build-mingw.sh` step.

Comment (vejbomar) [reply .github/scripts/toolchain/build-mingw-winpthreads.sh:24]: Thank you, for shedding some light on it. Resolving the thread, so it's not blocking merging

**Pull request merged**